coverage/docs/design.md

```diff
@@ -0,0 +1,156 @@
+# Overview
+This code coverage tool calculates per file, per package and overall coverage on target directories. It generates the following artifacts
+  - code coverage profile, which is produced by "[go test -coverprofile](https://blog.golang.org/cover)" and contains all block level code coverage data
+  - xml file that stores file-level and package-level code coverage, formatted to be readable by TestGrid
+
+The tool has two major modes of operation, based on whether it is running in post-submit or pre-submit workflow. Post-submit workflow starts when a commit to the base branch happens, e.g. when a pull request is merged. 
+Pre-submit workflow starts when a pull requested is created or updated with new commit.
+
+The tool performs the following additional operations when running in pre-submit workflow 
+  - after running code coverage on target directories,  it compares the new result with the one stored by 
+  the post-submit workflow and generate coverage difference. It reports coverage changes to the 
```

Comment:
Github post is the only consumer of the coverage change. I think it make more sense to put them together

Reply:
Sure, but the comment isn't only related to the coverage diff. On the next bullet point you reference it again when talking about the line coverage link.

This doesn't matter too much, it just seems strange that an important part of the behavior is mentioned as part of *two* other steps, rather than its own.